tests/languages/go/tests.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	appwrite "github.com/appwrite/sdk-for-go/appwrite"
+)
+
+func main() {
+	stringInArray := []interface{}{"string in array"}
+
+	client := appwrite.NewClient(10 * time.Second)
+	client.AddHeader("Origin", "http://localhost")
+	fmt.Print("\n\nTest Started\n")
+	testFooService(client, stringInArray)
+	testBarService(client, stringInArray)
+	testGeneralService(client, stringInArray)
+}
+
+func testFooService(client appwrite.Client, stringInArray []interface{}) {
+	foo := appwrite.NewFoo(client)
+	// Foo Service
+	response, err := foo.Get("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("foo.Get => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = foo.Post("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("foo.Post => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = foo.Put("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("foo.Put => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = foo.Patch("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("foo.Patch => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = foo.Delete("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("foo.Delete => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+}
+
+func testBarService(client appwrite.Client, stringInArray []interface{}) {
+	bar := appwrite.NewBar(client)
+	// Bar Service
+	response, err := bar.Get("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("bar.Get => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = bar.Post("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("bar.Post => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = bar.Put("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("bar.Put => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = bar.Patch("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("bar.Patch => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+
+	response, err = bar.Delete("string", 123, stringInArray)
+	if err != nil {
+		fmt.Printf("bar.Delete => error %v", err)
+	}
+	fmt.Printf("%s\n", response["result"])
+}
+
+func testGeneralService(client appwrite.Client, stringInArray []interface{}) {
+	general := appwrite.NewGeneral(client)
```

Comment:
Can we avoid `NewGeneral` and instead use `General` and same for other Appwrite services?

Reply:
This is an idiomatic factory pattern in go. (e.g. see https://www.sohamkamani.com/golang/2018-06-20-golang-factory-patterns/) 
The question is, if one want to harm the language convention in favor of the then "similar" SDK API.

Reply:
What do you think @lohanidamodar  ?

Reply:
@phaus we don't want to harm the language convention 👍🏻